syntax_and_semantics/structs.md

```diff
@@ -11,9 +11,30 @@ struct Point
 end
 ```
 
-## Passing
+Structs inherit from [Value](https://crystal-lang.org/api/Value.html) so they are allocated on the stack and passed by value: when you pass it to methods, return it from methods or assign it to variables, a copy of the value is actually passed (while classes inherit from [Reference](https://crystal-lang.org/api/Reference.html), are allocated on the heap and passed by reference). Therefore you should prefer using structs for immutable data types and/or stateless wrappers of other types, usually for performance reasons to avoid lots of small memory allocations when passing small copies might be more efficient (for more details, see the [performance guide](https://crystal-lang.org/docs/guides/performance.html#use-structs-when-possible)).
 
-A struct is [passed by value](http://crystal-lang.org/api/Value.html) while a class is passed by reference.
+Mutable structs are still allowed, but you should be careful when writing code involving mutability if you want to avoid surprises that are described below.
+
+## Passing by value
+
+A `struct` is _always_ passed by value, even when you return `self` from the method of that `struct`:
```

Comment:
"struct" shouldn't have the code markdown here because you don't talk about the keyword `struct`, you talk about a struct.
```suggestion
A struct is _always_ passed by value, even when you return `self` from the method of that struct:
```